test/discord.test.ts

```diff
@@ -293,161 +294,407 @@ describe('calling getUserId', () => {
+            expect(messagesSent).toEqual(1);
+        });
+
+        it('should call send message when creating task fails', done => {
+            expect.assertions(2);
+
+            const firstMessage = 'Creating task...';
+            const secondMessage = new RichEmbed().setTitle('error');
+            let messagesSent = 0;
+
+            const client = setupClient();
+
+            const commandControllerMock = new CommandControllerMockBuilder()
+                .withCreateTask(jest.fn(() => Promise.resolve(some(secondMessage))))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withCommandController(commandControllerMock)
+                .withClient(client)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent('!tasks create new task')
+                .withSend(jest.fn(async value => {
+                    messagesSent++;
+
+                    if (messagesSent === 1) {
+                        expect(value).toEqual(firstMessage);
+                    }
+                    if (messagesSent === 2) {
+                        expect(value).toEqual(secondMessage);
+                        done();
+                    }
+                }))
+                .build();
+
+            client.emit('message', message);
+        });
+
+        it('should send warning message when channel type is not text' , () => {
+            const client = setupClient();
+
+            const discordController = new DiscordControllerBuilder()
+                .withClient(client)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent('!tasks create new task')
+                .withChannelType('voice')
+                .build();
+
+            client.emit('message', message);
+
+            expect(message.channel.send)
+                .toBeCalledWith('!tasks create command must be called from a text channel' 
+                    + ' associated with a project');
+        });
+
+        it('should send error message and log when error getting project ID', done => {
+            expect.assertions(2);
+                
+            const client = setupClient();
+
+            const projectId = 0;
+            const error = 'error';
+            const channelName = 'channel';
+
+            let sentMessageValue = '';
+
+            const loggerMock = new LoggerMockBuilder()
+                .withWarn(jest.fn().mockImplementation(value => {
+                    expect(value).toBe(`Error creating task: ${error}`);
+                    expect(sentMessageValue).toBe(`Unable to find ActiveCollab project for channel ` 
+                        + channelName);
+                    done();
+                }))
+                .build();
+
+            const mappingControllerMock = new MappingControllerMockBuilder()
+                .withGetProjectId(jest.fn(() => { throw Error(error); }))
```

Comment:
`Error` is a constructor function so needs to be called with `new` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error

Also unnecessary parentheses. Should be `jest.fn(() => throw new Error(error))`.